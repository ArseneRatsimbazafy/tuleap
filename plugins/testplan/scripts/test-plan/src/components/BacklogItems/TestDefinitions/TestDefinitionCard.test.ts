/*
 * Copyright (c) Enalean, 2020 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import { shallowMount } from "@vue/test-utils";
import TestDefinitionCard from "./TestDefinitionCard.vue";
import { TestDefinition } from "../../../type";

describe("TestDefinitionCard", () => {
    it("Display a test definition as a card", () => {
        const wrapper = shallowMount(TestDefinitionCard, {
            propsData: {
                test_definition: {
                    id: 123,
                    short_type: "test_def",
                    summary: "Test definition summary",
                } as TestDefinition,
            },
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    it("Display an icon for automated tests", () => {
        const wrapper = shallowMount(TestDefinitionCard, {
            propsData: {
                test_definition: {
                    id: 123,
                    short_type: "test_def",
                    summary: "Test definition summary",
                    automated_tests: "Automated test name",
                },
            },
        });

        expect(wrapper.find("[data-test=automated-test-icon]").exists()).toBe(true);
    });
});
