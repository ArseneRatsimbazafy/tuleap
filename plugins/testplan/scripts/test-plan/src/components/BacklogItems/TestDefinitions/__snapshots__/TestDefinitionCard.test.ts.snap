// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestDefinitionCard Display a test definition as a card 1`] = `
<div
  class="tlp-card test-plan-test-definition-card"
>
  <div
    class="test-plan-test-definition-xref-title"
  >
    <span
      class="test-plan-test-definition-xref"
    >
      
            test_def #123
        
    </span>
    
        Test definition summary
    
  </div>
   
  <div
    class="test-plan-test-definition-icons"
  >
    <!---->
     
    <test-definition-card-status-stub
      test_definition="[object Object]"
    />
  </div>
</div>
`;
